src/Parser/Visitor/DocBlockTypeAnnotations.php

```diff
@@ -52,9 +53,23 @@ private function extractDocBlocks(array $docBlocks, Context $context): void
 
             foreach ($docBlock->getTags() as $tag) {
                 if (($tag instanceof Param || $tag instanceof Return_) && $tag->getType() !== null) {
-                    $type = $this->typeToFullyQualified($tag->getType(), $context);
-                    if ($type) {
-                        $this->namespaces[] = $type;
+                    $type = $tag->getType();
+                    $typesToResolve = [];
+
+                    if ($type instanceof Object_) {
+                        $typesToResolve[] = $type;
+                    }
+                    // To resolve generic definitions correctly we have to split the type into its original
+                    // type and the value part.
+                    if ($type instanceof Collection) {
+                        $typesToResolve[] = new Object_($type->getFqsen());
+                        $typesToResolve[] = $type->getValueType();
+                    }
```

Comment:
(This does not concernt your PR)

Reading the code I am fairly certain that it does not handle union types - both, in comments and actual type annotations (Coming in PHP8: https://wiki.php.net/rfc/union_types_v2 )

Reply:
It in fact does not. This should be a different issue, though.

Reply:
#15